File: src/utils/file_io.py
from os.path import exists

class Content:
    def __init__(self, path):
        self.path = path
        if not isinstance(self.path, str):
            print("ERROR: invalid path type.")
            exit(1)
        if not exists(self.path):
            print("ERROR: no such directory or file.")
            exit(1)
    
    def get_path(self):
        return self.path
    
    def get_content(self):
        try:
            content = open(self.path, "r").read()
        except:
            print("ERROR: something went wrong during file content reading.")
            exit(1)
        return content
    
__all__ = [
    "Content"
]

if __name__ == "__main__":
    import os
    raise Exception(f"`{os.path.basename(__file__)} can\'t be ran as `{__name__}`")

File: src/utils/fileIO.js
const process = require("process")
const fs = require("fs")

class Content {
    constructor(path) {
        this.path = path

        if (typeof(this.path) != "string") {
            console.error("ERROR: invalid path type.")
            process.exit(1)
        } 

        if (!fs.existsSync(this.path)) {
            console.error("ERROR: no such directory or file.")
            process.exit(1)
        }
    }
    
    get_content() {
        let content

        try {
            content = fs.readFileSync(this.path, "utf-8")
        } catch (error) {
            console.error("ERROR: something went wrong during file content reading.")
            process.exit(1)   
        }

        return content
    }
}

module.exports = {
    Content
}

if (require.main === module) {
    const path = require("path")

    console.error(`ERROR: \`${path.parse(__filename).name}\` file can\'t be ran as main.`)
    process.exit(1)
}